### DISCUSSION TITLE
Model form disappears on Validation Error

### DISCUSSION THREAD

**Discussion by alvins-dev** [wire-elements/modal#183] [Q&A]
@PhiloNL Thank you for the package.
I am stuck with the validation part when working with the modal.
```ruby
public function submit()
    {
        $this->validate();
        $this->user->save();
    }
```
After I click `submit` , Livewire is returning validation error when there is.
`serverMemo: {errors: {user.email: ["The email field is required."]}`
`@error('user.email') <p {{ $message }} </p> @enderror`
But when the modal is re-rendered, it removes the whole form-content leaving just LivewireUIModal HTML wrap.
How can I display the validation errors within Modal?
(1)
![1](https://user-images.githubusercontent.com/76688351/151026060-f12345d3-fb19-43e4-8faf-9a8042a3d9ec.png)
(2)
![2](https://user-images.githubusercontent.com/76688351/151026079-2c0f773a-809c-4483-855f-33888ef33f81.png)
(3)
![3](https://user-images.githubusercontent.com/76688351/151026096-5ae4b51e-8ee6-4c7d-a21a-15803b17eb91.png)
(4)
![4](https://user-images.githubusercontent.com/76688351/151026107-83c4f43c-b58a-4496-a5b5-2a937bb40415.png)

**Comment by PhiloNL** [comment]
This often happens if you have multiple root elements. Livewire only allows one root element to be present. So try wrapping the content of your component in a div.

**Comment by Mte90** [comment]
I have the same issue, I added validate on update and the rules variable.

I think that there is something in JS that removes the modal at all on clicking the submit button.

**Reply by PhiloNL** [reply]
In some cases, this was also caused because the form was missing `wire:submit.prevent="foo"` on the form. Also verify that there are no unclosed `divs`.

**Reply by Mte90** [reply]
This if you have a form but with this package `<x-modal form-action="update">` there is this and in the `<x-slot name="content">` there aren't any form.

I guess that I should write in my modal:

```
<button wire:submit.prevent="submit" wire:click="$emit('closeModal')">{{ __('messages.add') }}</button>
```

Or maybe it's better just to remove the wire:click method so I close it by PHP? But the event that execute the update method in the component with that should be executed right?

**Reply by PhiloNL** [reply]
If you use a blade component for the modal that includes the form, you can set the wire:submit.prevent by editing that specific component. Not sure if this is the cause of the issue, though. I would recommend checking for misplaced divs, which could also cause this issue as well.

**Reply by Mte90** [reply]
I was able to fix it in that way and removing the `wire:click`

**Comment by chiragparekh** [comment]
I had same issue, removing comment of the root element fixed the issue.
